playback_markers.js

```diff
@@ -0,0 +1,73 @@
+
+function collectPlaybackMarkersDuration(targetedName) {
+  let result = {};
+  const categoryMediaPlayback = 16;
+  for (let marker of window.wrappedJSObject.filteredMarkers) {
+    if (marker.category != categoryMediaPlayback) {
+      continue;
+    }
+    console.log(marker);
+    if (marker.name.indexOf(targetedName) == -1) {
+      continue;
+    }
+    const duration = marker.end - marker.start;
+    if (marker.name in result) {
+      result[marker.name].push(duration)
+    } else {
+      result[marker.name] = [ duration ];
+    }
+  }
+  return result;
+}
+
+(() => {
+  const markerNames = [
+    "RequestDecode", "RequestDemux", "RequestData",
+    "CopyDemuxedData", "CopyDecodedData",
+  ];
+  let results = {};
+  let shouldDisplayResult = false;
+  for (let name of markerNames) {
+    let dataSet = collectPlaybackMarkersDuration(name);
+    Object.entries(dataSet).forEach(([key, data]) => {
+      let result = {
+        markerName: name,
+        markerKey: key,
+        mean: 0,
+        median: 0,
+        stddev: 0,
+        variance: 0,
+      };
+
+      // Median
+      let sortedData = data.slice();
+      sortedData.sort((a, b) => a - b);
+      result.median = sortedData[Math.floor(sortedData.length / 2)];
+
+      // Average
+      let sum = 0;
+      for (let value of sortedData) {
+        sum += value;
+      }
+      result.mean = sum / sortedData.length;
+
+      // Variance
+      for (let value of sortedData) {
+        result.variance += Math.pow(value - result.mean, 2);
+      }
+      result.variance /= sortedData.length;
```

Comment:
```suggestion
      result.variance /= sortedData.length;
      result.stddev = Math.sqrt(result.variance);
```

The stddev is essential because it's homogenous to what we're measuring and can be used with mean. It's especially important because we don't display the histogram yet, so we're completely oblivious to distributions that are, e.g. bi-modal. In addition, testing on a 4k video, the variance / stddev is huge for non-keyframes [^0], we might want to have a look. 

Thankfully there is a split between key frames and other types of frames, so this expected bimodality is not a problem, but still.

You can read https://blog.paul.cx/post/profiling-firefox-real-time-media-workloads/#a-short-real-time-audio-performance-assessment-primer for a reason why this is important, but https://en.wikipedia.org/wiki/Anscombe's_quartet (linked from the article) graphically illustrate the point.

It should be fairly straightforward to add an histogram.

[^0]: mean at 10.4ms, variance at 15.15, so stddev at ~3.9ms, which means that ~30% of events are longer than `10.4 + 3.9 = 14.3ms`, which is dangerously close to the budget for a 60fps video.